diff --git a/scripts/headers_install.sh b/scripts/headers_install.sh
index 593f8879c641..49bf804eb2bb 100755
--- a/scripts/headers_install.sh
+++ b/scripts/headers_install.sh
@@ -31,7 +31,7 @@ trap 'rm -f "$OUTDIR/$FILE" "$OUTDIR/$FILE.sed"' EXIT
 for i in "$@"
 do
 	FILE="$(basename "$i")"
-	sed -E \
+	sed -r \
 		-e 's/([[:space:](])(__user|__force|__iomem)[[:space:]]/\1/g' \
 		-e 's/__attribute_const__([[:space:]]|$)/\1/g' \
 		-e 's@^#include <linux/compiler(|_types).h>@@' \
